tests/e2e/e2e.go

```diff
@@ -8,440 +8,231 @@ import (
-	subnet := []string{}
-	for i := 1; i <= int(networkConfig.NumValidators); i++ {
-		n := fmt.Sprintf("node%d", i)
-		subnet = append(subnet, n)
-	}
-	specs := []*rpcpb.BlockchainSpec{
-		{
-			VmName:      networkConfig.Name,
-			Genesis:     networkConfig.VMGenesisPath,
-			ChainConfig: networkConfig.VMConfig,
-			SubnetSpec: &rpcpb.SubnetSpec{
-				SubnetConfig: networkConfig.SubnetConfigPath,
-				Participants: subnet,
-			},
-		},
-	}
-
-	// Create subnet
-	createBlockchainsCtx, cancel := context.WithTimeout(ctx, 5*time.Minute)
-	sresp, err := anrCli.CreateBlockchains(
-		createBlockchainsCtx,
-		specs,
-	)
-	cancel()
-	require.NoError(err)
-
-	blockchainID = sresp.ChainIds[0]
-	networkID := sresp.ClusterInfo.NetworkId
-	subnetID := sresp.ClusterInfo.CustomChains[blockchainID].SubnetId
-	utils.Outf(
-		"{{green}}successfully added chain:{{/}} %s {{green}}subnet:{{/}} %s {{green}}participants:{{/}} %+v\n",
-		blockchainID,
-		subnetID,
-		subnet,
-	)
-	trackSubnetsOpt = runner_sdk.WithGlobalNodeConfig(fmt.Sprintf(`{"%s":"%s"}`,
-		config.TrackSubnetsKey,
-		subnetID,
-	))
-
-	require.NotEmpty(blockchainID)
-	require.NotEmpty(logsDir)
-
-	cctx, ccancel := context.WithTimeout(ctx, 2*time.Minute)
-	status, err := anrCli.Status(cctx)
-	defer ccancel()
-	require.NoError(err)
-	nodeInfos := status.GetClusterInfo().GetNodeInfos()
-
-	for _, nodeName := range subnet {
-		info := nodeInfos[nodeName]
-		u := fmt.Sprintf("%s/ext/bc/%s", info.Uri, blockchainID)
-		cli := rpc.NewJSONRPCClient(u)
-
-		// After returning healthy, the node may not respond right away
-		//
-		// TODO: figure out why
-		for i := 0; i < 10; i++ {
-			_, _, _, err = cli.Network(cctx)
-			if err != nil {
-				time.Sleep(1 * time.Second)
-				continue
-			}
-		}
-		require.NoError(err)
-
-		uris = append(uris, u)
-	}
-
-	network = &embeddedVMNetwork{
-		uris:      uris,
-		networkID: networkID,
-		chainID:   ids.FromStringOrPanic(blockchainID),
-	}
-}
+	network := fixture.NewTmpnetNetwork(owner, nodes, subnet)
+	return e2e.NewTestEnvironment(
+		e2e.NewTestContext(),
+		flagVars,
+		network,
+	).Marshal()
+}, func(envBytes []byte) {
+	// Run in every ginkgo process
+
+	// Initialize the local test environment from the global state
+	e2e.InitSharedTestEnvironment(ginkgo.GinkgoT(), envBytes)
+	blockchainID = e2e.GetEnv(e2e.NewTestContext()).GetNetwork().GetSubnet(vmName).Chains[0].ChainID
+})
 
 var _ = ginkgo.Describe("[HyperSDK APIs]", func() {
-	ctx := context.Background()
-	require := require.New(ginkgo.GinkgoT())
+	tc := e2e.NewTestContext()
+	require := require.New(tc)
 
 	ginkgo.It("Ping", func() {
-		workload.Ping(ctx, require, network.uris)
+		workload.Ping(tc.DefaultContext(), require, getE2EURIs(tc, blockchainID))
 	})
 
 	ginkgo.It("GetNetwork", func() {
-		workload.GetNetwork(ctx, require, network.uris, network.networkID, ids.FromStringOrPanic(blockchainID))
+		baseURIs := getE2EBaseURIs(tc)
+		baseURI := baseURIs[0]
+		client := info.NewClient(baseURI)
```

Comment:
Constructing info client here because `networkID` field on the constructed `tmpnet` is 0